integration/openshift/lib.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Contain helper functions and variables for the OpenShift integration
+# test scripts.
+#
+
+# ATTENTION: the caller of this lib should have the tests_dir exported
+#            correctly
+if [ ! -d "${tests_dir}" ]; then
+	echo "Unable to find the Kata Container tests directory."
+	exit 1
+fi
+
+source ${tests_dir}/.ci/lib.sh
+
+# The name of the Kubernetes runtimeClass resource for Kata Containers.
+#
+kata_runtimeclass=${kata_runtimeclass:-kata}
+
+# Build the openshift-tests binary.
+#
+# Return:
+#  The absolute path to the file.
+#
+function build_openshift_tests()
+{
+	local openshift_repo="github.com/openshift/origin"
+	local repo_dir="${GOPATH}/src/${openshift_repo}"
+	local cmd=""
+	{
+	[ -d "$repo_dir" ] || go get -d $openshift_repo
+	pushd "$repo_dir"
+		git checkout "release-$(get_version \
+			"externals.openshift.meta.newest-version")"
+		make WHAT=cmd/openshift-tests
+		cmd="$(realpath \
+			$(find -type f -executable -name "openshift-tests"))"
+	popd
+	} &>/dev/null
+	[ $? -eq 0 ] || return 1
+	echo $cmd
+}
+
+# Check if the runtimeClass resource is present in the cluster.
+#
+function is_runtimeclass_present()
+{
+	oc get runtimeclass/$kata_runtimeclass &>/dev/null
+}
+
+# Check if the Kata Containers admission controller is up and running so
+# that any spawned Pod will use the runtimeClass by default.
+#
+function is_kata_admission_controller_active()
+{
+	local pod="hello-openshift"
+
+	if ! is_runtimeclass_present; then
+		warn "The ${kata_runtimeclass} runtimeClass is not present"
+		return 1
+	fi
+
+	if oc get pod/${pod} &>/dev/null; then
+		warn "pod ${pod} is already created, cannot reliably check"
+		return 1
+	fi
+
+	# Create a hello world pod and check if it is using the runtimeClass.
+	#
+	curl -s https://raw.githubusercontent.com/openshift/origin/master/examples/hello-openshift/hello-pod.json |\
+		oc apply -f - &>/dev/null
+	local class_name=$(oc get -o jsonpath='{.spec.runtimeClassName}' \
+		pod/${pod})
+	oc delete pod/${pod} &>/dev/null
+	if [ "$class_name" != "$kata_runtimeclass" ]; then
+		warn "runtimeClass: expected ${kata_runtimeclass}, got ${class_name}"
+		return 1
+	fi
```

Comment:
I am not seeing a runtimeClass specified in the hello-pod.json.  I don't see how it can ever succeed the check to see if the class_name is equate to kata_runtimeclass.  Am I missing something?

Reply:
If the admission control is up and running correctly then regular pods (e.g. this hello-pod) will have their containers attached the Kata runtimeClass. In simple words, the admission controller will make almost all pods to use Kata runtime even if the pod descriptor doesn't say so.